src/types.rs

```diff
@@ -0,0 +1,12 @@
+pub enum FluentType {
+    FluentString(String),
+    FluentNone,
+    FluentNumber,
+}
+
+pub fn value_of(t: FluentType) -> String {
```

Comment:
You can implement traits on enum variants yet - https://internals.rust-lang.org/t/impl-trait-for-enum-variant/4131/3